### PR TITLE
Add support for retrieval with dependent types

### PR DESCRIPTION
**Pull request by kaneeldias**
## Purpose
$subject

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/4087

## Examples
```ballerina
public type EmployeeName record {|
    string firstName;
    string lastName;
|};

function employeeReadManyTest2() returns error? {
    RainierClient rainierClient = check new ();

    stream<EmployeeName, Error?> employeeStream = rainierClient->/employee.get();
    EmployeeName[] employees = check from EmployeeName employee in employeeStream 
        select employee;

    test:assertEquals(employees, [
        {firstName: "Tom", lastName: "Scott"},
        {firstName: "Jane", lastName: "Doe"},
        {firstName: "Hugh", lastName: "Smith"}
    ]);
    check rainierClient.close();
}
```

## Checklist
- [x] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: ballerina/tests/rainier_generated_client.bal [hunk 1]
```diff
@@ -91,14 +92,10 @@ public client class RainierClient {
...
-        if result is Error {
-            return new stream<Employee, Error?>(new EmployeeStream((), result));
-        } else {
-            return new stream<Employee, Error?>(new EmployeeStream(result));
-        }
-    }
+    isolated resource function get employee(EmployeeTargetType targetType = <>, string entity = "employee") returns stream<targetType, Error?> = @java:Method {
+        'class: "io.ballerina.stdlib.persist.Utils",
+        name: "queryStream"
+    } external;
```
[the remark this answers is not in the document]
Comment (daneshk) [reply ballerina/tests/rainier_generated_client.bal:98]: This is for temporary evaluation purposes. I don't think we can go ahead with the additional param. we need support from java binding to pass constant values. or add this via compiler modifier etc

[the remark this answers is not in the document]
Comment (niveathika) [reply ballerina/tests/rainier_generated_client.bal:98]: Can we modify a function definition from the plugin?

[the remark this answers is not in the document]
Comment (daneshk) [reply ballerina/tests/rainier_generated_client.bal:98]: Not sure. I doubt we can't.

File: ballerina/tests/employee-tests.bal [hunk 2]
```diff
@@ -128,7 +130,7 @@ function employeeCreateTestNegative() returns error? {
 function employeeReadOneTest() returns error? {
     RainierClient rainierClient = check new ();
 
-    Employee employeeRetrieved = check rainierClient->/employee/[employee1.empNo].get();
+    Employee employeeRetrieved = check rainierClient->/employeed/[employee1.empNo].get();
```
Comment (daneshk) [inline ballerina/tests/employee-tests.bal:133]:
```suggestion
    Employee employeeRetrieved = check rainierClient->/employee/[employee1.empNo].get();
```

**Pull request merged**